### PR TITLE
Ark chart improvements

### PR DESCRIPTION
**Pull request by The-Loeki**
@domcar & @unguiculus :
I've made several improvements to the Heptio Ark chart:
* use Helm's `crd-install` to remove the kubectl container workaround
* add logging/verbosity options
* optionally enable restic backups
* different providers for backups & pv's supported

I've noticed btw that in 0.10 they're going to change the CRD/configurations, FYI

### CODE REVIEW

**Comment by The-Loeki** [comment]
/assign @linki 

Is it OK to test?

**Comment by plumdog** [comment]
@The-Loeki this looks fairly similar to https://github.com/helm/charts/pull/7165 (which I should probably rebase).

I haven't checked whether they really do the same thing in the same way, or whether there are meaningful differences, but I'll have a look.

**Comment by The-Loeki** [comment]
@plumdog I'm sorry I didn't pick up on your earlier commit, but considering neither @linki, @domcar,  @unguiculus, @mattfarina or @viglesiasce can be bothered to reply to either your PR or mine, I'm going to hold of doing a thing until somebody starts responding ;)

**Comment by domcar** [comment]
@The-Loeki I didn't mean to be rude by not answering, unfortunately I don't think i'll have the time right now. Anyone else is available? @unguiculus ?

**Comment by domcar** [comment]
@The-Loeki I didn't mean to be rude by not answering, unfortunately I don't think i'll have the time right now. Anyone else is available? @unguiculus ?

**Comment by domcar** [comment]
@The-Loeki I didn't mean to be rude by not answering, unfortunately I don't think i'll have the time right now. Anyone else is available? @unguiculus ?

**Comment by domcar** [comment]
@The-Loeki i just noticed that your chart install the version 0.9.4, maybe you can update it to the newest version? It should be 0.9.8

File: stable/ark/templates/config.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+backupStorageProvider:
+  name: {{ .name }}
+  bucket: {{ .bucket  }}
+{{- with .config }}
+  config:
+  {{- with .region }}
+    region: {{ . }}
+  {{- end }}
+  {{- with .s3ForcePathStyle }}
+    s3ForcePathStyle: {{ . }}
```
Comment (domcar) [inline stable/ark/templates/config.yaml:35]: This should be quoted
 s3ForcePathStyle: {{ . | quote }}

File: stable/ark/templates/config.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  {{- end }}
+  {{- with .apitimeout }}
+    apiTimeout: {{ . }}
+  {{- end }}
+{{- end }}
+{{- end }}
+{{- with .backupStorageProvider }}
+backupStorageProvider:
+  name: {{ .name }}
+  bucket: {{ .bucket  }}
```
Comment (domcar) [inline stable/ark/templates/config.yaml:28]: you should add `resticLocation` here

**Comment by domcar** [comment]
@The-Loeki in `templates/_helpers.tpl` I think you don't need anymore the `hookSA` correct?

**Pull request closed without merging**